src/pyk/kcfg/explore.py

```diff
@@ -354,33 +355,52 @@ def _check_abstract(self, node: KCFG.Node, kcfg: KCFG) -> bool:
         kcfg.create_cover(node.id, new_node.id)
         return True
 
+    def _check_terminal(self, node: KCFG.Node) -> bool:
+        _LOGGER.info(f'Checking terminal: {shorten_hashes(node.id)}')
+        if self.kcfg_semantics.is_terminal(node.cterm):
+            _LOGGER.info(f'Terminal node: {shorten_hashes(node.id)}.')
+            return True
+        return False
```

Comment:
```suggestion
    def _check_terminal(self, node: KCFG.Node) -> bool:
        _LOGGER.info(f'Checking terminal: {node.id}')
        if self.kcfg_semantics.is_terminal(node.cterm):
            _LOGGER.info(f'Terminal node: {node.id}.')
            return True
        return False
```

Node IDs are `int` now, I can open a PR changing the rest.

Reply:
Ah, I see, ok